libs/promises/api/celix/DefaultScheduledExecutor.h

```diff
@@ -108,9 +111,12 @@ namespace celix {
         void wait() override {
             bool done = false;
             while (!done) {
-                std::lock_guard<std::mutex> lck{mutex};
+                std::unique_lock lck{mutex};
```

Comment:
should be lock_guard instead of unique_lock (left over from some reverted refactoring)